test/e2e/apps/rc.go

```diff
@@ -558,3 +563,55 @@ func updateReplicationControllerWithRetries(c clientset.Interface, namespace, na
 	}
 	return rc, pollErr
 }
+
+type rcPatchFunc func(pdb *v1.ReplicationController) ([]byte, error)
+
+func patchRCOrDie(cs kubernetes.Interface, dc dynamic.Interface, ns string, name string, f rcPatchFunc, subresources ...string) (updated *v1.ReplicationController, err error) {
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		old := getRCStatusOrDie(dc, ns, name)
+		patchBytes, err := f(old)
+		if updated, err = cs.CoreV1().ReplicationControllers(ns).Patch(context.TODO(), old.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, subresources...); err != nil {
+			return err
+		}
+		return nil
+	})
+
+	framework.ExpectNoError(err, "Waiting for the rc update to be processed in namespace %s", ns)
```

Comment:
i think the annotation here should use past tense and mention what action failed:

> "failed to patch the rc %q in namespace %q", name, ns

i'd also use ReplicationController instead of rc, but looks like the old code uses rc.

Reply:
Updated in 0f303d9d841